consensus/tendermint/core/tendermint_start_round_test.go

```diff
@@ -0,0 +1,124 @@
+package core
+
+import (
+	"context"
+	"github.com/clearmatics/autonity/common"
+	"github.com/clearmatics/autonity/consensus/tendermint/committee"
+	"github.com/clearmatics/autonity/core/types"
+	"github.com/golang/mock/gomock"
+	"math/big"
+	"testing"
+)
+
+func prepareCommittee() types.Committee {
+	// prepare committee.
+	member1 := types.CommitteeMember{
+		Address:     common.HexToAddress("0x01234567890"),
+		VotingPower: new(big.Int).SetInt64(1),
+	}
+	member2 := types.CommitteeMember{
+		Address:     common.HexToAddress("0x01234567891"),
+		VotingPower: new(big.Int).SetInt64(1),
+	}
+	member3 := types.CommitteeMember{
+		Address:     common.HexToAddress("0x01234567892"),
+		VotingPower: new(big.Int).SetInt64(1),
+	}
+	member4 := types.CommitteeMember{
+		Address:     common.HexToAddress("0x01234567892"),
+		VotingPower: new(big.Int).SetInt64(1),
+	}
+	return types.Committee{member1, member2, member3, member4}
+}
+
+func generateBlock(height *big.Int) *types.Block {
+	header := &types.Header{Number: height}
+	block := types.NewBlock(header, nil, nil, nil)
+	return block
+}
+
+// It test the page-6, line-11, StartRound() function from proposer point of view of tendermint pseudo-code.
+// Please refer to the algorithm from here: https://arxiv.org/pdf/1807.04938.pdf
+func TestTendermintProposerStartRound(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	currentCommittee := prepareCommittee()
+	committeeSet, err := committee.NewSet(currentCommittee, currentCommittee[3].Address)
+	if err != nil {
+		t.Error(err)
+	}
+
+	currentHeight := big.NewInt(10)
+	currentBlock := generateBlock(currentHeight)
+	proposalHeight := big.NewInt(11)
+	proposalBlock := generateBlock(proposalHeight)
+	clientAddr := currentCommittee[0].Address
+
+	backendMock := NewMockBackend(ctrl)
+	backendMock.EXPECT().Address().AnyTimes().Return(clientAddr)
+	backendMock.EXPECT().LastCommittedProposal().AnyTimes().Return(currentBlock, currentCommittee[3].Address)
+	backendMock.EXPECT().Committee(proposalHeight.Uint64()).AnyTimes().Return(committeeSet, nil)
+	backendMock.EXPECT().SetProposedBlockHash(proposalBlock.Hash()).AnyTimes()
+	backendMock.EXPECT().Broadcast(context.Background(), committeeSet, gomock.Any()).AnyTimes().Return(nil)
+	backendMock.EXPECT().Sign(gomock.Any()).AnyTimes().Return(nil, nil)
+
+	// create consensus core.
+	c := New(backendMock)
+	c.pendingUnminedBlocks[proposalHeight.Uint64()] = proposalBlock
+	c.committeeSet = committeeSet
+	c.sentProposal = false
+	c.height = currentHeight
+	round := int64(0)
+	c.startRound(context.Background(), round)
+
+	if c.sentProposal != true {
+		t.Error("Proposer did not send proposal on consensus start round")
+	}
```

Comment:
```suggestion
assert.True(t, c.sentProposal, "Proposer did not send proposal on consensus start round")
```

We could use the testify assert package here just to make our tests a bit more streamlined, it contains a lot of matchers beyond just `assert.True`.